server/db/model/Event.js

```diff
@@ -1,4 +1,65 @@
-/* eslint-disable */
 var mongoose = require('../index.js'),
-    config = require('../../../config/default.js'),
-    Schema = mongoose.Schema;
+    Schema = mongoose.Schema,
+    Location = require('./Location.js');
+
+// Define the document Schema
+var schema = new mongoose.Schema({
+    title: String,
+    desc: String,
+    startDate: {
+        type: Date,
+        default: Date.now()
+    },
+    endDate: {
+        type: Date,
+        default: Date.now()
+    },
+    category: {
+        type: String,
+        enum: ['General', 'Food', 'Tech Talk', 'Sponsor Event', 'Other'],
+        default: 'General'
+    },
+    location: {
+        type: Schema.Types.ObjectId
+    }
+});
+
+// Allow us to query by name
+schema.query.byName = function(name) {
+    return this.findOne({
+        name: new RegExp(name, 'i')
+    });
+};
+
+schema.query.byLocation = function(name) {
+    return Location.find()
```

Comment:
Rather than importing the Location object, can you reference it with `mongoose.model('Location').find()`, that way we don't potentially introduce any circular dependencies between model schemas